src/mesa/util/os_time.c

```diff
@@ -54,14 +54,21 @@
 #  error Unsupported OS
 #endif
 
+#if defined(CLOCK_MONOTONIC_RAW) /* linux */
+#define MANGOHUD_CLOCKTYPE CLOCK_MONOTONIC_RAW
+#elif defined(CLOCK_MONOTONIC_FAST) /* freebsd */
+#define MANGOHUD_CLOCKTYPE CLOCK_MONOTONIC_FAST
+#else /* fallback */
+#define MANGOHUD_CLOCKTYPE CLOCK_MONOTONIC
+#endif
 
 int64_t
 os_time_get_nano(void)
 {
 #if DETECT_OS_LINUX || DETECT_OS_BSD
 
    struct timespec tv;
-   clock_gettime(CLOCK_MONOTONIC, &tv);
+   clock_gettime(MANGOHUD_CLOCKTYPE, &tv);
```

Comment:
Technically speaking you should verify the return value, if it maybe returns `ENOTSUP` fallback to `CLOCK_MONOTONIC` or report an error or crash. I am not 100% sure it is worth it tho, as `CLOCK_MONOTONIC_RAW` is supported since kernel 2.6.28.

Reply:
It's not worth it: nobody is using a kernel from 2008.